src/slow_hash_plug.c

```diff
@@ -209,28 +202,107 @@ int cn_slow_hash(const void *data, size_t length, char *hash, void *memory)
 	 * next address  <-+
 	 */
 	if (have_aesni)
+#if 1 && __AVX2__ /* or actually __AES__ && __AVX__ && __BMI__ */
+	__asm__(
+		"vmovdqa %0,%%xmm0\n\t"
+		"vmovdqa %1,%%xmm1\n\t"
+		"1:\n\t"
+		"vmovd %%xmm0,%%eax\n\t"
+		"andnl %%eax,%4,%%ecx\n\t"
+		"vmovdqa (%%rbx,%%rcx),%%xmm2\n\t"
+		"vaesenc %%xmm0,%%xmm2,%%xmm2\n\t"
+		"vpxor %%xmm2,%%xmm1,%%xmm1\n\t"
+		"vmovq %%xmm2,%%rax\n\t"
+		"vmovdqa %%xmm1,(%%rbx,%%rcx)\n\t"
+		"andnl %%eax,%4,%%ecx\n\t"
+		"mulq (%%rbx,%%rcx)\n\t"
+		"vmovq %%rdx,%%xmm3\n\t"
+		"vpinsrq $1,%%rax,%%xmm3,%%xmm3\n\t"
+		"vpaddq %%xmm3,%%xmm0,%%xmm0\n\t"
+		"vpxor (%%rbx,%%rcx),%%xmm0,%%xmm3\n\t"
+		"vmovdqa %%xmm0,(%%rbx,%%rcx)\n\t"
+		"vmovd %%xmm3,%%eax\n\t"
+		"andnl %%eax,%4,%%ecx\n\t"
+		"vmovdqa (%%rbx,%%rcx),%%xmm1\n\t"
+		"vaesenc %%xmm3,%%xmm1,%%xmm1\n\t"
+		"vpxor %%xmm1,%%xmm2,%%xmm2\n\t"
+		"vmovq %%xmm1,%%rax\n\t"
+		"vmovdqa %%xmm2,(%%rbx,%%rcx)\n\t"
+		"andnl %%eax,%4,%%ecx\n\t"
+		"mulq (%%rbx,%%rcx)\n\t"
+		"vmovq %%rdx,%%xmm0\n\t"
+		"vpinsrq $1,%%rax,%%xmm0,%%xmm0\n\t"
+		"vpaddq %%xmm0,%%xmm3,%%xmm3\n\t"
+		"vpxor (%%rbx,%%rcx),%%xmm3,%%xmm0\n\t"
+		"vmovdqa %%xmm3,(%%rbx,%%rcx)\n\t"
+		"decl %%esi\n\t"
+		"jnz 1b\n\t"
+		:
+		: "m" (a), "m" (b), "b" (long_state), "S" (ITER / 4), "D" (~(uint32_t)(MEMORY - AES_BLOCK_SIZE))
+		: "ax", "cx", "dx", "xmm0", "xmm1", "xmm2", "xmm3", "memory", "cc");
```

Comment:
Note to self: I need to add `si` to clobbered registers (here, and ditto below).

Reply:
Unexpectedly, listing `si` as clobbered turned out non-trivial - getting `error: ‘asm’ operand has impossible constraints` because the input expression for this register is a constant. Will probably need to use a temporary variable.